### ISSUE TITLE
Optimize import statements

### ISSUE THREAD

**Issue by EBatTiVo** [TiVo/intellij-haxe#10]
In Haxe source files, add missing import statements and remove unused ones.  New code shall be generated at the top of the file, right after the opening comment and before any other definitions.

TIR: STB-2902

**Comment by yanhick** [comment]
@EBatTiVo, the import statement is added but it is added at the very top of the class, above the 'package' statement, and this triggers an error in Intellij and it won't compile.

**Comment by sganapavarapu1** [comment]
Yes, this caused compilation errors to me as well. This needs to be fixed to add import statement after the package statement.

**Comment by EBatTiVo** [comment]
Issue #104 is a duplicate of the unfinished work being tracked here.

**Comment by as3boyan** [comment]
Confirmed

**Comment by as3boyan** [comment]
Comments below `package` "break" it

**Comment by EBatTiVo** [comment]
How so?  findFirstNotNullChild() is finding the comment?  Or is insertAfter() not working?

I'm just curious, but I can wait for the fix.   8)

**Comment by as3boyan** [comment]
The problem is caused by `UsefulPsiTreeUtil.isWhitespaceOrComment` function, which fails.

**Comment by EBatTiVo** [comment]
Oh, really?  Is the element type incorrect for a comment that comes after some code?

**Comment by as3boyan** [comment]
Fixed

**Comment by as3boyan** [comment]
Of course - that is because PsiJavaToken:DOC_COMMENT doesn't match PsiWhitespace nor PsiComment

```haxe
public static boolean isWhitespaceOrComment(PsiElement element) {
    if (element instanceof PsiJavaToken) {
      PsiJavaToken psiJavaToken = (PsiJavaToken)element;
      IElementType tokenType = psiJavaToken.getTokenType();
      return tokenType.equals(HaxeTokenTypeSets.DOC_COMMENT) || tokenType.equals(HaxeTokenTypeSets.MML_COMMENT) || tokenType.equals(HaxeTokenTypeSets.MSL_COMMENT);
    }

    return element instanceof PsiWhiteSpace || element instanceof PsiComment;
  }
```

**Comment by EBatTiVo** [comment]
OK.  Just submit.  I'll update the jar later today.

**Comment by EBatTiVo** [comment]
> Of course - that is because PsiJavaToken:DOC_COMMENT doesn't match PsiWhitespace nor PsiComment

Ah, my fault then.

**Comment by as3boyan** [comment]
And not only DOC_COMMENT, MSL_COMMENT/MML_COMMENT too

**Comment by EBatTiVo** [comment]
Pretty sure this was finally fixed with 3ecd486e88bdd63e9f70bb8a1d6714ef2c44bd1f.
